packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -200,7 +200,7 @@ const devHtmlHook: IndexHtmlTransformHook = async (
     // and ids are properly handled
     const validPath = `${htmlPath}${trailingSlash ? 'index.html' : ''}`
     proxyModulePath = `\0${validPath}`
-    proxyModuleUrl = wrapId(proxyModulePath)
+    proxyModuleUrl = joinUrlSegments(base, wrapId(proxyModulePath))
```

Comment:
Maybe we could move the `joinUrlSegments` out of these two branches so it is more clear that this always need to be added.

```js
proxyModuleUrl = joinUrlSegments(base, proxyModuleUrl)
```